lib/Doctrine/Common/Annotations/AnnotationReader.php

```diff
@@ -107,6 +107,8 @@ class AnnotationReader implements Reader
         // PHP CodeSniffer
         'codingStandardsIgnoreStart' => true,
         'codingStandardsIgnoreEnd' => true,
+        // PHPStan
+        'template' => true, 'implements' => true, 'extends' => true,
```

Comment:
These annotations are too widespread to be ignored, specifically talking about existing applications that rely on things such as `@template("mystuff.twig")`, for example.

Usage of `@phpstan-template` (already ignored by the parser) is to be endorsed instead

Reply:
If the `@template` annotation was autoloaded (which it should be), there's no BC break to be found here.

Reply:
also, these ignore rules are case sensitive according to the comment above.

Due to PSR-4 autoloading being as case-sensitive as your filesystem (meaning it is, for most servers), the code probably uses `@Template("mystuff.twig")`

Reply:
Chatted with @nicolas-grekas about this and he doesn't see a problem for Symfony: the `@Template` annotation is "kinda deprecated", and even then it would continue to work as long as users probably autoload this.

Of course, other applications still might be affected, but all the change would do is that we would no longer error if the annotation wasn't autoloader before.

Reply:
This is a general problem, not framework specific.

If there is really not a problem with it, then I'd need a test showing a working `@template` annotation with this patch applied (or an existing test)

Reply:
Following up, we have a fixture class that includes multiple `@Template` annotations: https://github.com/doctrine/annotations/blob/346c8bc45a85a53012a29f86e2b11322c0679f21/tests/Doctrine/Tests/Common/Annotations/Fixtures/Controller.php#L16. This file continues to be parsed correctly. Obviously, aliasing any annotation as `@template` would cause this to be an issue, but since the exclude list is case-sensitive I don't think this should be too big a problem.